scripts/release/strip-appimage-graphics-libs.sh

```
@@ -328,6 +328,72 @@ validate_sharun_lib_path() {
+  fi
+
+  # Check if the file is a shell script (not an ELF binary).
+  local first_bytes
+  first_bytes="$(LC_ALL=C head -c 2 "$apprun" 2>/dev/null || true)"
+  if [ "$first_bytes" = $'\x7fE' ]; then
+    # AppRun is an ELF binary — cannot patch with sed.
+    return 1
+  fi
+
+  # Idempotency guard: skip if we already patched this AppRun.
+  if grep -q 'cd.*"\$APPDIR"' "$apprun" 2>/dev/null; then
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)